src/screens/Instructor/InstPlaylist/controllers/setup.js

```diff
@@ -120,7 +125,14 @@ class InstPlaylistSetup extends StateController {
       this.setOffering(offering);
     }
 
-    api.contentLoaded();
+    // check if the user is the course admin for this playlist
+    const instIdx = _.findIndex(this.offering.instructorIds, { id: user.userId });
+    if (instIdx < 0) {
+      // back to student version playlist page if not authorized
+      window.location = links.offeringDetail(this.offering.id, playlist.id);
```

Comment:
Are there any subtle differences, or are these all interchangeable here: `this.offering.id === playlist.offeringId === this.playlist.offeringId === offeringId`

And `playlist.id === this.playlist.id === this.playlistId`?

Reply:
They are basically the same in this function, the function `this.setPlaylist(playlist)` will assign `playlist` to `this.playlist`, similiar for offering.

And since the offering data is got via `playlist.offeringId`, so `playlist.offeringId === this.offering.id`

The `this.playlistId` is got from the URL when the page is loaded.

Reply:
I added Rui's comment as a code comment to this function